src/modules/m_restrictchans.cpp

```diff
@@ -21,19 +21,44 @@
 
 
 #include "inspircd.h"
+#include "modules/account.h"
 
 typedef insp::flat_set<std::string, irc::insensitive_swo> AllowChans;
 
 class ModuleRestrictChans : public Module
 {
 	AllowChans allowchans;
+	bool allowregistered;
+
+	bool CanCreateChannel(LocalUser* user, const std::string& name)
+	{
+		const AccountExtItem* accountext = GetAccountExtItem();
+		if (allowregistered && accountext->get(user))
```

Comment:
GetAccountExtItem() may return NULL if the services_account module is not loaded.

```suggestion
		if (allowregistered && accountext && accountext->get(user))
```